### ISSUE TITLE
[feature] Renaming link text in markdown

### ISSUE THREAD

**Issue by etiennekintzler** [vimwiki/vimwiki#1138]
Is there way a command/way to rename the link text for markdown link ? I know the existence of `<Leader>wr` but it renames the link URL, not the link text.

**Comment by etiennekintzler** [comment]
> Vimwiki is manly used for links managment from disctance, so we will not add a mapping to spare a ci[. Well we can but it is not so usefull as it wont spare types (user must place cursor on the link anyway and press 2 keys at least).

This is not meant exactly what I meant. 

Let's say you have an index.wiki with a link `[description](link_to_file.md)`. You open this file in a new buffer (using <CR> for instance). Once this file is open you can already rename the link (that is `link_to_file.md`) with the command `:VimwikiRenameFile`. This idea here is to have a command to rename the link text (that is `description`) from the opened file (and not from the index.wiki).

The tldr; is to have a command like `:VimwikiRenameFile`  to rename the link text.

**Comment by tinmarino** [comment]
@etiennekintzler Tjere is definitely no command like that.
The best would be to enhence rename file to take a second optional argument.

**Comment by deafpolygon** [comment]
Using `[[ ]]` links does not get renamed either. Only `[]()` Markdown style links (the link in the `()` get renamed).  But if I link to a  .md file in vimwiki using e.g. `[[Some Link]]` then `<CR>` to the file.  `Some Link.md` gets created, I'm able to save it.  If I  rename it using `wr`, say yes and change the name to `Useful Link` - the file gets renamed to `Useful Link.md`.  That works as expected, going back to the linked page and it still has `[[Some Link]]` and going back in it re-creates `Some Link.md`.  

Is there a way to fix this? I'd like all link references using `[[ ]]` style to also be updated.

**Comment by Gamer-Kold** [comment]
Will this get merged in? What else needs to be done?
